### PR TITLE
React Tooltips

### PR DESCRIPTION
**Pull request by AndrwM**
## Description
Convert our Sage Tooltips to a React flavored version. These tooltips don't have a wrapping DOM element and pass mouse events to components that it wraps.

Thanks to @pixelflips for working out the original tooltip positioning math. 🤘

### Screenshots
image

## Test notes
See the storybook for example: 
http://localhost:6006/?path=/story/sage-tooltip--default

- Tooltip should Portal _outside_ of the trigger DOM location
- Check to ensure React Component is perfomant

## Related
BUILD-367

### CODE REVIEW

File: packages/sage-react/lib/Tooltip/Tooltip.story.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import { centerXY } from '../story-support/decorators';
+import Tooltip from './Tooltip';
+import Button from '../Button';
+
+storiesOf('Sage/Tooltip', module)
+  .addDecorator(withKnobs)
+  .addDecorator(centerXY)
+  .add('Default', () => (
+    <>
+      <br/><br/><br/><br/><br/><br/><br/><br/>
```
Comment (philschanely) [inline packages/sage-react/lib/Tooltip/Tooltip.story.jsx:13]: Perhaps accomplish this with a local style on a wrapper `div`?

File: packages/sage-react/lib/Tooltip/TooltipElement.jsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+};
+
+TooltipElement.defaultProps = {
+  position: TOOLTIP_POSITIONS.DEFAULT,
+  size: TOOLTIP_SIZES.DEFAULT,
+  theme: TOOLTIP_THEMES.DEFAULT,
+};
+
+TooltipElement.propTypes = {
+  parentDomRect: PropTypes.object.isRequired,
```
Comment (philschanely) [inline packages/sage-react/lib/Tooltip/TooltipElement.jsx:77]: Mind alphabetizing?

File: packages/sage-react/lib/Tooltip/TooltipElement.jsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import classnames from 'classnames';
+
+import {
+  TOOLTIP_SIZES,
+  TOOLTIP_POSITIONS,
+  TOOLTIP_THEMES,
+} from './configs';
+
+const TooltipElement = ({
+  parentDomRect,
```
Comment (philschanely) [inline packages/sage-react/lib/Tooltip/TooltipElement.jsx:12]: Mind alphabetizing?

File: packages/sage-react/lib/Tooltip/Tooltip.jsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+const Tooltip = ({
+  children,
+  ...rest
+}) => {
+  const [active, setActive] = useState(false);
+  const [parentDomRect, setParentDomRect] = useState(null);
+
+  function handleMouseEnter(evt) {
+    console.log(evt.target.getBoundingClientRect());
```
Comment (philschanely) [inline packages/sage-react/lib/Tooltip/Tooltip.jsx:21]: Just a reminder to pull these logs.

**Review by philschanely — APPROVED** [review]
Looking good, @AndrwM !

**Review by JulaB — APPROVED** [review]
👍 This looks great!

**Comment by QuintonJason** [comment]
A11y consideration: the tooltip doesn't open if it is tabbed into, it's working correctly with a mouse interaction

File: packages/sage-react/lib/Tooltip/TooltipElement.jsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        top = parentDomRect.top - tooltipRef.current.offsetHeight - distance;
+        left = parentDomRect.left + (parentDomRect.width - tooltipRef.current.offsetWidth) / 2;
+      break;
+    }
+
+    setCoordinates({top: top, left: left});
+  }, [tooltipRef.current]);
+
+  return (
+    <div ref={tooltipRef} className={classNames} style={coordinates}>
```
Comment (QuintonJason) [inline packages/sage-react/lib/Tooltip/TooltipElement.jsx:64]: A11y consideration: add `role=tooltip` to trigger assistive tech to get `aria-live="polite"` for free

Comment (AndrwM) [reply packages/sage-react/lib/Tooltip/TooltipElement.jsx]: @QuintonJason Thanks! Added focus and a11y html attrs.

**Pull request merged**